### PR TITLE
PLENTY-314 fix website completely breaking down when shop key is not set

### PR DESCRIPTION
**Pull request by RimantasAdeoWeb**

### CODE REVIEW

File: tests/Validators/PluginOrderValidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+        $this->assertEquals($expectedResult, $this->validatorMock->validate());
+    }
+
+    public function validationTestDataProvider(): array
+    {
+        return [
+            'It is valid when IO is loaded before Findologic in the current plugin set' => [
+                [
+                    (object) [
```
Comment (TheKeymaster) [inline tests/Validators/PluginOrderValidatorTest.php:77]: Why are these casted into raw objects? You could use a mock, which is created inside of the test instead.

Comment (RimantasAdeoWeb) [reply tests/Validators/PluginOrderValidatorTest.php:77]: For convenience and saving time while writing these tests. Sure, I could use a mock, but would it be somehow better in any way?

File: src/Validators/MainValidator.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * @var LoggerContract
+     */
+    private $logger;
+
+    /**
+     * @return bool
+     */
+    public function validate()
+    {
+        if ($this->validationStatus !== null) {
```
[the remark this answers is not in the document]
Comment (RimantasAdeoWeb) [reply src/Validators/MainValidator.php:29]: Does it? Lame. Do you know how I would get a singleton then?

File: tests/Validators/PluginOrderValidatorTest.php (hunk 1 above)
Comment (RimantasAdeoWeb) [reply tests/Validators/PluginOrderValidatorTest.php:77]: Nevermind, found the type. Ok, I'll make these into mocks then

File: src/Validators/MainValidator.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheKeymaster) [reply src/Validators/MainValidator.php:29]: We create two singletons in `\Findologic\Providers\FindologicServiceProvider::register`:

https://github.com/findologic/plugin-plentymarkets-ceres-api/blob/9559159d3706a724bb279c181c334e9b00dfb682/src/Providers/FindologicServiceProvider.php#L23

File: tests/Validators/PluginOrderValidatorTest.php (hunk 1 above)
Comment (TheKeymaster) [reply tests/Validators/PluginOrderValidatorTest.php:77]: It is overall easier to change, and I personally prefer mocks over some random objects floating around.

File: src/Validators/MainValidator.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (RimantasAdeoWeb) [reply src/Validators/MainValidator.php]: Yup, I did that as well

**Review by TheKeymaster — APPROVED** [review]

**Pull request merged**